src/useMintUsd.ts

```diff
@@ -0,0 +1,105 @@
+import { useMutation } from '@tanstack/react-query';
+import type { ethers } from 'ethers';
+import { fetchMintUsd } from './fetchMintUsd';
+import { fetchMintUsdWithPriceUpdate } from './fetchMintUsdWithPriceUpdate';
+import { fetchPriceUpdateTxn } from './fetchPriceUpdateTxn';
+import { useImportContract } from './useImports';
+import { useSynthetix } from './useSynthetix';
+
+interface CollateralType {
```

Comment:
we either need to export this type (and then I'd argue this file is not the right place for it as it is fairly commonly used)
```suggestion
export interface CollateralType {
```

Or what I reckon is better - not to declare a separate type. Just list all properties as is as the query return type.

Same comment for other PRs.

Reply:
ohhhhh damn, of course passing redundant data to this hook and also describing such a large type is redundant. Passing only the address is enough.

I don't understand why I did this.

This certainly needs to be fixed.

Thanks.